### PR TITLE
Gloo Mesh Enterprise v1.1.0 Upgrade docs

### PR DESCRIPTION
**Pull request by ashleywang1**

### CODE REVIEW

File: docs/content/operations/upgrading.md [hunk 1]
```diff
@@ -1,104 +1,56 @@
...
-that these steps will provide a comprehensive, zero-downtime upgrade experience. To reiterate:
-
 {{< notice warning >}}
-This upgrade process is **not guaranteed** to provide a comprehensive, zero-downtime upgrade experience.
+This upgrade process is for Gloo Mesh Enterprise, from v1.1.0-betax to v1.1.0.
 
-While Gloo Mesh is under initial development and in a pre-1.0.0 state, the recommended way to access new functionality
-is via a fresh install of Gloo Mesh.
+If you have an older version installed, the recommended way to access new functionality
+is via a fresh install of Gloo Mesh Enterprise.
```
Comment (artberger) [inline docs/content/operations/upgrading.md:12]: Does this mean uninstall and reinstall? If so, maybe rephrase:
```suggestion
For previous versions, uninstall and re-install Gloo Mesh Enterprise, such as described in one of the [setup guides]({{% versioned_link_path fromRoot="/getting_started//#deploying-gloo-mesh" %}}).
```

File: docs/content/operations/upgrading.md [hunk 2]
```diff
@@ -1,104 +1,56 @@
...
 {{< /notice >}}
 
-1\. Following the steps outlined by the [Getting Started Guide]({{% versioned_link_path fromRoot="/getting_started/" %}}),
-download the version of `meshctl` corresponding to the version of Gloo Mesh to which you would like to upgrade.
-
-2\. Scale down both the `networking` and `discovery` deployments on the Gloo Mesh management cluster to zero replicas.
-This will prevent each component from processing resources structured in a way not expected by the new versions of each
-component.
+1\.
```
Comment (artberger) [inline docs/content/operations/upgrading.md:15]: Maybe describe why they do this, like:
```suggestion
1\. In your terminal, set environment variables for your current and target Gloo Mesh Enterprise versions to use in the subsequent steps.
```

File: docs/content/operations/upgrading.md [hunk 3]
```diff
@@ -1,104 +1,56 @@
...
 
 ```shell
-kubectl scale deployment -n gloo-mesh networking --replicas 0
+# Set your current version here
+CURRENT_VERSION=v1.1.0-beta35
+# Add your desired version here
+UPGRADE_VERSION=v1.1.0
 ```
 
+2\. Upgrade CRDs on your management cluster and all of your data plane clusters:
```
Comment (artberger) [inline docs/content/operations/upgrading.md:24]: Do they have to switch the kube contexts to each individual cluster to run these commands?
```suggestion
2\. Upgrade the Gloo Mesh Enterprise CRDs on your management cluster and all of your data plane clusters. You can change contexts before applying the CRDs by running `kubectl config set-context <cluster-name>`.
```

File: docs/content/operations/upgrading.md [hunk 4]
```diff
@@ -1,104 +1,56 @@
...
+k apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/discovery.mesh.gloo.solo.io_v1_crds.yaml
+k apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/multicluster.solo.io_v1alpha1_crds.yaml
+k apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/networking.enterprise.mesh.gloo.solo.io_v1beta1_crds.yaml
+k apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/networking.mesh.gloo.solo.io_v1_crds.yaml
+k apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/observability.enterprise.mesh.gloo.solo.io_v1_crds.yaml
+k apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/rbac.enterprise.mesh.gloo.solo.io_v1_crds.yaml
+k apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/settings.mesh.gloo.solo.io_v1_crds.yaml
```
Comment (artberger) [inline docs/content/operations/upgrading.md:33]: I know this process might change, but maybe just spell out `kubectl` in case they don't have a shorthand set.
```suggestion
kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/admin.enterprise.mesh.gloo.solo.io_v1alpha1_crds.yaml
kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/discovery.mesh.gloo.solo.io_v1_crds.yaml
kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/multicluster.solo.io_v1alpha1_crds.yaml
kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/networking.enterprise.mesh.gloo.solo.io_v1beta1_crds.yaml
kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/networking.mesh.gloo.solo.io_v1_crds.yaml
kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/observability.enterprise.mesh.gloo.solo.io_v1_crds.yaml
kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/rbac.enterprise.mesh.gloo.solo.io_v1_crds.yaml
kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/settings.mesh.gloo.solo.io_v1_crds.yaml
```

File: docs/content/operations/upgrading.md [hunk 5]
```diff
@@ -1,104 +1,56 @@
...
-and CRDs are updated to a version compatible with the version of Gloo Mesh you are upgrading to. Refer to the 
-[setup guide on registering a cluster]({{% versioned_link_path fromRoot="/setup/register_cluster" %}})
-and be sure to use the same cluster names and contexts that were used at the initial cluster registration time.
-
-8\. Scale the `discovery` deployment to one replica, and wait for all `discovery` resources such as `meshes`, `destinations`,
-and `workloads` to be written to the management cluster. This may take a few minutes, and will ensure that the `networking`
-component has access to all the data it needs to continue processing user-provided network configuration. Discovery is
-complete when the pod no longer outputs a steady stream of logs or when all expected resources can be found on the cluster.
 
+4\. Helm Upgrade for the Management Plane
```
Comment (artberger) [inline docs/content/operations/upgrading.md:43]: Maybe 
```suggestion
4\. Set your Kubernetes context to the management plane cluster, and upgrade the Helm installation.
```

File: docs/content/operations/upgrading.md [hunk 6]
```diff
@@ -1,104 +1,56 @@
...
+helm upgrade --install gloo-mesh --namespace gloo-mesh \
+  'https://storage.googleapis.com/gloo-mesh-enterprise/gloo-mesh-enterprise/gloo-mesh-enterprise-1.1.0.tgz' \
+  --set licenseKey=$LICENSE_KEY
 ```
 
-9\. Scale the `networking` deployment to one replica. Errors may be propagated to various `networking` resources such as
-`trafficpolicies`, `virtualmeshes`, `failoverservices`, and `accesspolicies` as it starts watches on remote clusters,
-but it will reach a steady state after a few moments.
-
+5\. Helm Upgrade for the Data Planes
```
Comment (artberger) [inline docs/content/operations/upgrading.md:50]:
```suggestion
5\. For each data plane cluster, set your Kubernetes context to the cluster, and upgrade the Helm installation.
```

File: docs/content/operations/upgrading.md [hunk 7]
```diff
@@ -1,104 +1,56 @@
...
 ```shell
-kubectl scale deployment -n gloo-mesh networking --replicas 1
+helm upgrade --install enterprise-agent --namespace gloo-mesh https://storage.googleapis.com/gloo-mesh-enterprise/enterprise-agent/enterprise-agent-1.1.0.tgz
 ```
 
-10\. Run `meshctl check` to verify that all resources are in a healthy state. If not, check the logs on the `discovery`
-and `networking` pods as well as the `status` fields on unhealthy resources to begin debugging. Refer to our 
+6\. Run `meshctl check` on the management to verify that all resources are in a healthy state. Refer to our
```
Comment (artberger) [inline docs/content/operations/upgrading.md:55]: Are there any prereqs they have to do to get `meshctl check` to work? It doesn't work for me.
```suggestion
6\. Set the Kubernetes context to the management plane cluster and then check that your Gloo Mesh Enterprise resources are in a healthy state. Refer to our
```

File: docs/content/operations/upgrading.md [hunk 8]
```diff
@@ -1,104 +1,56 @@
...
-The following steps describe a general approach to navigating the Gloo Mesh upgrade process for versions of Gloo Mesh
-< 1.0.0. This process involves scaling down the Gloo Mesh control plane while various components and CRDs are updated,
-but should not introduce downtime in service-to-service communication or policy enforcement because Gloo Mesh does not
-operate in the request path and the underlying service meshes (e.g. Istio) are not impacted by this process. That said,
-while Gloo Mesh is in its initial development ahead of general availability (i.e. pre-1.0.0), there is **no guarantee**
-that these steps will provide a comprehensive, zero-downtime upgrade experience. To reiterate:
-
 {{< notice warning >}}
-This upgrade process is **not guaranteed** to provide a comprehensive, zero-downtime upgrade experience.
+This upgrade process is for Gloo Mesh Enterprise, from v1.1.0-betax to v1.1.0.
```
Comment (artberger) [inline docs/content/operations/upgrading.md:9]: Would this upgrade process cause any downtime, like the deployments would go down or other types of impact? If so, could we call that out?

Comment (ashleywang1) [reply docs/content/operations/upgrading.md:9]: This _should_ not cause any downtime, given that we are rolling over the management plane, and that does not affect traffic. @ilackarms is this correct, or should we still add a warning?

File: docs/content/operations/upgrading.md (hunk 4 above)
Comment (ashleywang1) [reply docs/content/operations/upgrading.md:33]: Oh yeah absolutely - we definitely should never use `k` in our docs.

File: docs/content/operations/upgrading.md (hunk 7 above)
Comment (ashleywang1) [reply docs/content/operations/upgrading.md:55]: They need to have a `meshctl` version that is v1.1.0 or higher.

File: docs/content/operations/upgrading.md [hunk 9]
```diff
@@ -1,104 +1,63 @@
...
+# Set your current version here (Must be a v1.1.0-betax version)
+CURRENT_VERSION=v1.1.0-beta35
+# Add your desired version here (Must be a v1.1.x version)
+UPGRADE_VERSION=v1.1.0
 ```
 
+2\. Upgrade the Gloo Mesh CRDs on your management cluster and all of your data plane clusters. You can change contexts before applying the CRDs by running `kubectl config set-context <cluster-name>`.
 ```shell
-kubectl scale deployment -n gloo-mesh discovery --replicas 0
+kubectl apply -f https://raw.githubusercontent.com/solo-io/gloo-mesh/$UPGRADE_VERSION/install/helm/gloo-mesh-crds/crds/admin.enterprise.mesh.gloo.solo.io_v1alpha1_crds.yaml
```
Comment (ashleywang1) [inline docs/content/operations/upgrading.md:25]: Caveat: we actually don't want to use the $UPGRADE_VERSION here (It's fine for the special v1.1.0 case because our releases line up correctly).

We actually want to use the `gloo-mesh` dependency version in the gloo-mesh-enterprise release that we are upgrading to. This is information that is only available in the `go.mod` file (in a private repository) or by searching through the changelogs (should be available in the docs, but is currently broken) for the latest Dependency Bump for gloo-mesh.

File: docs/content/operations/upgrading.md (hunk 8 above)
Comment (Sodman) [reply docs/content/operations/upgrading.md:9]: The data plane stays up but you can have some down time of the management plane, which could result in delays of new configs being applied, which could potentially cause some downtime in certain situations.

I think we should keep the "not guaranteed zero downtime" disclaimer for now - but true zero downtime upgrades will be a high priority item for the next release. Happy to discuss further internally, if needed!

Comment (artberger) [reply docs/content/operations/upgrading.md:9]: ok, how about:

```suggestion
This upgrade process is for Gloo Mesh Enterprise, from v1.1.0-betax to v1.1.0. During the upgrade, the data plane continues to run, but you might not be able to modify the configurations through the management plane. Because zero downtime is not guaranteed, try testing the upgrade in a staging environment before upgrading your production environment.
```

Comment (Sodman) [reply docs/content/operations/upgrading.md:9]: Perfect!

File: docs/content/operations/upgrading.md [hunk 10]
```diff
@@ -1,104 +1,63 @@
...
-while Gloo Mesh is in its initial development ahead of general availability (i.e. pre-1.0.0), there is **no guarantee**
-that these steps will provide a comprehensive, zero-downtime upgrade experience. To reiterate:
-
 {{< notice warning >}}
-This upgrade process is **not guaranteed** to provide a comprehensive, zero-downtime upgrade experience.
+This upgrade process is for Gloo Mesh Enterprise, from v1.1.0-betax to v1.1.0. During the upgrade, the data plane continues to run, but you might not be able to modify the configurations through the management plane. Because zero downtime is not guaranteed, try testing the upgrade in a staging environment before upgrading your production environment.
 
-While Gloo Mesh is under initial development and in a pre-1.0.0 state, the recommended way to access new functionality
-is via a fresh install of Gloo Mesh.
+If you are on a previous version, uninstall and re-install Gloo Mesh Enterprise, as described in one of the [setup guides]({{% versioned_link_path fromRoot="/getting_started//#deploying-gloo-mesh" %}}).
```
Comment (artberger) [inline docs/content/operations/upgrading.md:11]: I saw in the preview that I messed up the link :( 
```suggestion
If you are on a previous version, uninstall and re-install Gloo Mesh Enterprise, as described in one of the [setup guides]({{% versioned_link_path fromRoot="/getting_started/#deploying-gloo-mesh" %}}).
```

**Review by artberger — APPROVED** [review]
